cmd/agent-smoke/main.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"os"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	smoke "github.com/grafana/agent/cmd/agent-smoke/internal"
+)
+
+func main() {
+	var (
+		debugLog          bool
+		namespace         string
+		kubeconfig        string
+		withTimeout       time.Duration
+		chaosFrequency    time.Duration
+		mutationFrequency time.Duration
+	)
+
+	flag.BoolVar(&debugLog, "debug", false, "enable debug logging")
+	flag.StringVar(&namespace, "namespace", "agent-smoke-test", "namespace smoke test should run in")
+	flag.StringVar(&kubeconfig, "kubeconfig", "", "absolute path to the kubeconfig file")
+	flag.DurationVar(&withTimeout, "with-timeout", time.Duration(0), "timeout after duration, example 3h")
```

Comment:
Timeout verbiage feels off, maybe `smoke-test-duration`?

Reply:
I think `timeout` is common terminology but I'm not opposed to changing this to just `duration` perhaps. `smoke-test-duration` seems kind of redundant.